### PR TITLE
Electron Support

### PR DESCRIPTION
**Pull request by James-Oswald**

### CODE REVIEW

File: src/electron/electron.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+// eslint-disable-next-line
+const {app, BrowserWindow} = require("electron");
+const path = require("path");
+const url = require("url");
+
+function createWindow() {
+    // Create the browser window.
+    const win = new BrowserWindow({
+        width: 800,
```
Comment (10dine) [inline src/electron/electron.js:15]: Currently this size causes components to be squished. Please increase the width to 1200 or more.

File: package.json [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
-  "main": "index.js",
+  "main": "build/index.html",
   "scripts": {
     "lint": "gts lint",
     "fix": "gts fix",
     "build": "vite build --mode production",
     "build-dev": "vite build --mode development",
-    "live-edit": "vite serve --open",
     "build-docs": "typedoc",
+    "build-electron": "vite build --mode electron && electron-packager build peircemyheart",
```
Comment (10dine) [inline package.json:12]: Please add an overwrite script -
`"build-electron-ow": "vite build --mode electron && electron-packager build peircemyheart --overwrite"`

**Review by 10dine — APPROVED** [review]
The directory and file changes might cause issues with some people if they're already working on their assigned issues. However, I'm not sure about this. Also, prettier is failing the electron.js file, might have to make a .prettierignore file.

File: package.json (hunk 2 above)
Comment (James-Oswald) [reply package.json:12]: What does this do? Vite will be default overwrite the build folder, is the folder with the electron app compiled by build not being overwritten?

File: src/electron/electron.js (hunk 1 above)
Comment (James-Oswald) [reply src/electron/electron.js:15]: Should be fixed on the renderer (CSS) side. Will be handled by #42.

**Comment by James-Oswald** [comment]
> The directory and file changes might cause issues with some people if they're already working on their assigned issues. 

This should be resolved since I used `git mv` to move the source files rather than copy pasting them into new directories, preserving their git history. The build system has been updated in `vite.config.js` to take this change of structure into account for normal builds. 

> Also, prettier is failing the electron.js file, might have to make a .prettierignore file.

How did this pass lint status check than? What setup are you using to get these errors?

File: package.json (hunk 2 above)
Comment (10dine) [reply package.json:12]: Some modules do get overwritten. From my testing, when I changed the window size on electron.js it did not get applied. So, I'm assuming core electron.js files don't get overwritten when doing regular build.

Comment (James-Oswald) [reply package.json:12]: Thanks, good to know.

**Comment by 10dine** [comment]
> @10dine
> 
> > The directory and file changes might cause issues with some people if they're already working on their assigned issues.
> 
> This should be resolved since I used `git mv` to move the source files rather than copy pasting them into new directories, preserving their git history. The build system has been updated in `vite.config.js` to take this change of structure into account for normal builds.

This is pretty cool.

> > Also, prettier is failing the electron.js file, might have to make a .prettierignore file.
> 
> How did this pass lint status check than? What setup are you using to get these errors?

image

I used the script you made `npm run lint` and almost all error were from prettier. Unclear on if I did something wrong.

**Comment by James-Oswald** [comment]
This is quite strange, all is well in terms of linting on my end. 
image

**Comment by 10dine** [comment]
> This is quite strange, all is well in terms of linting on my end. image

It seems to be an issue on my end.

**Review by DawnTheWitch — APPROVED** [review]
Praise the Emperor and his holy code

**Pull request merged**